pymc4/distributions/continuous.py

```diff
@@ -1,23 +1,189 @@
+
+    Parameters
+    ----------
+    mu : float|tensor
+        Mean.
+    sigma : float|tensor
+        Standard deviation (sigma > 0).
+
+    Examples
+    --------
+    .. code-block:: python
+        @pm.model
+        def model():
+            x = pm.Normal('x', mu=0, sigma=10)
+
+    Developer Notes
+    ---------------
+    Parameter mappings to TensorFlow Probability are as follows:
+
+    - mu: loc
+    - sigma: scale
+    """
+
+    def __init__(self, name, mu, sigma, **kwargs):
+        super().__init__(name, mu=mu, sigma=sigma, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        mu, sigma = conditions["mu"], conditions["sigma"]
+        return tfd.Normal(loc=mu, scale=sigma)
+
+
+class HalfNormal(PositiveContinuousDistribution):
+    r"""Half-normal random variable.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x \mid \tau) =
+           \sqrt{\frac{2\tau}{\pi}}
+           \exp\left(\frac{-x^2 \tau}{2}\right)
+       f(x \mid \sigma) =\sigma
+           \sqrt{\frac{2}{\pi}}
+           \exp\left(\frac{-x^2}{2\sigma^2}\right)
+
+    .. note::
+
+       The parameters ``sigma``/``tau`` (:math:`\sigma`/:math:`\tau`) refer to
+       the standard deviation/precision of the unfolded normal distribution, for
+       the standard deviation of the half-normal distribution, see below. For
+       the half-normal, they are just two parameterisation :math:`\sigma^2
+       \equiv \frac{1}{\tau}` of a scale parameter
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(0, 5, 200)
+        for sigma in [0.4, 1., 2.]:
+            pdf = st.halfnorm.pdf(x, scale=sigma)
+            plt.plot(x, pdf, label=r'$\sigma$ = {}'.format(sigma))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ==========================================
+    Support   :math:`x \in [0, \infty)`
+    Mean      :math:`\sqrt{\dfrac{2}{\tau \pi}}` or :math:`\dfrac{\sigma \sqrt{2}}{\sqrt{\pi}}`
+    Variance  :math:`\dfrac{1}{\tau}\left(1 - \dfrac{2}{\pi}\right)` or :math:`\sigma^2\left(1 - \dfrac{2}{\pi}\right)`
+    ========  ==========================================
+
+    Parameters
+    ----------
+    sigma : float
+        Scale parameter :math:`sigma` (``sigma`` > 0) (only required if ``tau`` is not specified).
+
+    Examples
+    --------
+    .. code-block:: python
+
+        @pm.model
+        def model():
+            x = pm.HalfNormal('x', sigma=10)
+
+    Developer Notes
+    ---------------
+    Parameter mappings to TensorFlow Probability are as follows:
+
+    - sigma: scale
+    """
+
+    def __init__(self, name, sigma, **kwargs):
+        super().__init__(name, sigma=sigma, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
```

Comment:
What's the point of these status methods if all they do is forward to the tfp distributions? I especially ask because in the next PR I'm going to remove the argument name differences so then they'll be exactly the same?

Reply:
For now, it was so that tests keep passing without changing them. From a larger point of view, I like that `unpack_conditions`, `_init_distribution`, and `_init_transform` provide developer hooks into initializing a random variable (the `Distribution.__init__` is below to provide context).

I might suggest keeping this, since then we can push out custom distributions without getting them into tensorflow probability (`HalfStudentT`, for example).

```python
        self.conditions = self.unpack_conditions(**kwargs)
        self._distribution = self._init_distribution(self.conditions)
        self.plate = plate
        super().__init__(
            self.unpack_distribution, name=name, keep_return=True, keep_auxiliary=False
        )
        if name is None and observed is not None:
            raise ValueError(
                "Observed variables are not allowed for anonymous (with name=None) Distributions"
            )
        self.model_info.update(observed=observed)
        self.transform = self._init_transform(transform)
        if self.plate is not None:
            self._distribution = tfd.Sample(self._distribution, sample_shape=self.plate)
```

Reply:
Keeping test working I understand

But for future case even in the above example though couldn't this line be 

```
self._distribution = tfd.some_dist(self.conditions)
```
if all that's happening is were initializing a tfd distribution?

Reply:
yep! that's even reasonable as a default. this is a bit of over engineering, but I think it may come in handy: 
- it is nice to get a raw `tfp` object to debug what is going on with a model using `pm.Normal._init_distribution(whatever)`
- there might be some shape handling that has to go on
- `self.conditions` might have some extra arguments
- `tfd` might not implement the distribution

Reply:
Er, to elaborate: I feel strongly that it should stay for this PR. I'm actually ok with removing it after merging this - you're right that it gets rid of some indirection and makes the intention more clear. I'd prefer it stayed for the above reasons, but I *do* recognize it is a little over-engineered!